converter/convertBIN2csv.m

```diff
@@ -23,34 +30,46 @@
 
 %% Reading file
 while read
-    packet = parsePacket(fid);
+    packet = parseBtPacket(fid);
     switch packet.type
         case 'orn'
             ORN.data = cat(2, ORN.data, packet.orn);
             ORN.timestamp = cat(2, ORN.timestamp, packet.timestamp);
-            
         case {'eeg4', 'eeg8'}
+            nSample = size(packet.data, 2);
+            t = linspace(packet.timestamp, packet.timestamp + ...
+                (nSample - 1)/device_info.data_rate, nSample);
             EEG.data = cat(2, EEG.data, packet.data);
-            EEG.timestamp = cat(2,EEG.timestamp, repmat(packet.timestamp,...
-                1, size(packet.data, 2)));
-            
+            EEG.timestamp = cat(2,EEG.timestamp, t);
+        case 'marker_event'
+            Marker.timestamp = cat(1, Marker.timestamp, packet.timestamp);
+            Marker.code = cat(1, Marker.code, packet.code);
         case 'env'
             ENV.temperature = cat(2, ENV.temperature, packet.temperature);
             ENV.light = cat(2, ENV.light, packet.light);
             ENV.battery = cat(2, ENV.battery, packet.battery);
             ENV.timestamp = cat(2, ENV.timestamp, packet.timestamp);
-            
         case 'ts'
             TS = cat(2, TS, packet.ts);
-            
+        case 'dev_info'
+            if exist('device_info', 'var') && ~strcmp(packet.adc_mask, device_info.adc_mask)
+                display(['Old ADC mask: ' device_info.adc_mask]);
+                display(['New ADC mask: ' packet.adc_mask]);
+                warning(AdcMaskWarningMsg);
+                break
+            end
+            device_info = packet;
+        case 'disconnect'
+            timestamp = packet;
```

Comment:
apologies  - again I don't understand. `timestamp = packet;` ?? what does that mean?

Reply:
It was a special packet called Timestamp without a real usage but later its purpose changed and now FW saves this packet whenever there is a BT disconnect event. I'll change `timestamp` to `disconnect`.